### ISSUE TITLE
Installation (of helper tool) fails

### ISSUE THREAD

**Issue by GnrGnr** [halo/LinkLiar#13]
I unfortunately was unable to get the new release to install properly. After running LinkLiar.app, the password entry window shows up briefly but disappears again automatically after a second.

**Comment by halo** [comment]
Which OS X version? Can you run the `Console` app to see if there are any logs? What do you have in  System Preferences -> Security -> Allow apps downloaded from -> ?

**Comment by GnrGnr** [comment]
OS X 10.10.2; I usually like to get notified when new apps install, so current setting is "Only Apps downloaded from Mac App Store and Identi ..."

There was an entry in the console:

15/02/15 19:26:29,832 ReportCrash[8031]: com.apple.message.domain: com.apple.crashreporter.writereport.crash
com.apple.message.signature: Link
com.apple.message.signature2: com.funkensturm.Link ||| 1.0 (1.0.1)
com.apple.message.signature3: 1EB0177CBA024D8986DDE5063139359E
com.apple.message.result: NO
com.apple.message.summarize: YES
SenderMachUUID: 8C15E826-C574-3656-8CA9-235A0AC8C3BB

I am not a developer. So I just learned a couple of things :)

**Comment by halo** [comment]
Bummer, this is when I wish there should be more logging available to the end-users haha. I use the same OS X and settings so that's not the problem.

Is there any crash report file related to LinkLiar in this directory? `~/Library/Logs/CrashReporter`

If everything fails, I'll have to implement some sort of "logging on demand" and/or wait for more feedback by other users to see if this is a general problem or something which primarily occurs on your system. I assume something with thread safety (i.e. code executing in parallel and stepping on each other's toes) so I can try to secure that area of the code a little bit more, too. Because it's really weird that the superuser password window comes up but then suddenly disappears...

**Comment by GnrGnr** [comment]
Unfortunately there is nothing in ~/Library/Logs/CrashReporter apart from some crash reports of my iDevices.

I thought it might just be enough time to enter my PW in the SU-window while it is open. I managed to enter a couple of characters before it closes (copy/paste does not work). 

If i managed to enter at least parts of my PW, i am kind of stuck "somewhere". It seems like the password window is still in focus (even though it is not visible), as i can not switch to other applications (by clicking on them directly or using CMD-TAB) for a couple of seconds.

**Comment by halo** [comment]
Would you mind trying this version while looking at the `Console`?

https://github.com/halo/LinkLiar/releases/download/v1.0.1/LinkLiarDebug.app.zip

I took away some shaky thread assertions and now full logging is activated.

**Comment by GnrGnr** [comment]
Here's what it says:

15/02/15 21:03:37,736 Link[8714]: An uncaught exception was raised
15/02/15 21:03:37,736 Link[8714]: *** -[NSArray objectsAtIndexes:]: index 2 in index set beyond bounds [0 .. 1]
15/02/15 21:03:37,736 Link[8714]: (
	0   CoreFoundation                      0x00007fff8e85d66c __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff934df76e objc_exception_throw + 43
	2   CoreFoundation                      0x00007fff8e792933 -[NSArray objectsAtIndexes:] + 995
	3   Link                                0x0000000102f73aec Link + 105196
	4   Link                                0x0000000102f6f7e1 Link + 88033
	5   Link                                0x0000000102f6a4a1 Link + 66721
	6   Link                                0x0000000102f6a417 Link + 66583
	7   Link                                0x0000000102f6a1f4 Link + 66036
	8   Link                                0x0000000102f6a088 Link + 65672
	9   Link                                0x0000000102f76ab3 Link + 117427
	10  CoreFoundation                      0x00007fff8e76b85f -[NSSet makeObjectsPerformSelector:] + 223
	11  AppKit                              0x00007fff8f4035cd -[NSIBObjectData nibInstantiateWithOwner:options:topLevelObjects:] + 1216
	12  AppKit                              0x00007fff8ef13605 loadNib + 384
	13  AppKit                              0x00007fff8f4845ab +[NSBundle(NSNibLoading) _loadNibFile:nameTable:options:withZone:ownerBundle:] + 313
	14  AppKit                              0x00007fff8ef12cd5 -[NSBundle(NSNibLoading) loadNibNamed:owner:topLevelObjects:] + 201
	15  AppKit                              0x00007fff8ef12aa1 +[NSBundle(NSNibLoading) loadNibNamed:owner:] + 344
	16  AppKit                              0x00007fff8ef0e549 NSApplicationMain + 605
	17  libdyld.dylib                       0x00007fff8c1725c9 start + 1
)
15/02/15 21:03:37,737 Link[8714]: *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[NSArray objectsAtIndexes:]: index 2 in index set beyond bounds [0 .. 1]'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff8e85d66c __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff934df76e objc_exception_throw + 43
	2   CoreFoundation                      0x00007fff8e792933 -[NSArray objectsAtIndexes:] + 995
	3   Link                                0x0000000102f73aec Link + 105196
	4   Link                                0x0000000102f6f7e1 Link + 88033
	5   Link                                0x0000000102f6a4a1 Link + 66721
	6   Link                                0x0000000102f6a417 Link + 66583
	7   Link                                0x0000000102f6a1f4 Link + 66036
	8   Link                                0x0000000102f6a088 Link + 65672
	9   Link                                0x0000000102f76ab3 Link + 117427
	10  CoreFoundation                      0x00007fff8e76b85f -[NSSet makeObjectsPerformSelector:] + 223
	11  AppKit                              0x00007fff8f4035cd -[NSIBObjectData nibInstantiateWithOwner:options:topLevelObjects:] + 1216
	12  AppKit                              0x00007fff8ef13605 loadNib + 384
	13  AppKit                              0x00007fff8f4845ab +[NSBundle(NSNibLoading) _loadNibFile:nameTable:options:withZone:ownerBundle:] + 313
	14  AppKit                              0x00007fff8ef12cd5 -[NSBundle(NSNibLoading) loadNibNamed:owner:topLevelObjects:] + 201
	15  AppKit                              0x00007fff8ef12aa1 +[NSBundle(NSNibLoading) loadNibNamed:owner:] + 344
	16  AppKit                              0x00007fff8ef0e549 NSApplicationMain + 605
	17  libdyld.dylib                       0x00007fff8c1725c9 start + 1
)
15/02/15 21:03:37,761 com.apple.xpc.launchd[1]: (com.apple.ReportCrash[8730]) Endpoint has been activated through legacy launch(3) APIs. Please switch to XPC or bootstrap_check_in(): com.apple.ReportCrash
15/02/15 21:03:37,889 SecurityAgent[8728]: Failed to initialize IconCache named: com.apple.iconservices with error: Error Domain=NSCocoaErrorDomain Code=4097 "Couldn’t communicate with a helper application." (connection to service named com.apple.iconservices) UserInfo=0x7fd1c2c548f0 {NSDebugDescription=connection to service named com.apple.iconservices}
15/02/15 21:03:37,890 SecurityAgent[8728]: Error returned from iconservicesagent: Error Domain=NSCocoaErrorDomain Code=4097 "Couldn’t communicate with a helper application." (connection to service named com.apple.iconservices) UserInfo=0x7fd1c2e33660 {NSDebugDescription=connection to service named com.apple.iconservices}
15/02/15 21:03:38,278 com.apple.xpc.launchd[1]: (com.apple.xpc.launchd.oneshot.0x10000023.Link[8714]) Service exited due to signal: Abort trap: 6

15/02/15 21:03:38,309 authd[82]: engine[8714]: evaluate returned -60008 returning errAuthorizationInternal
15/02/15 21:03:38,364 ReportCrash[8730]: Saved crash report for Link[8714] version 1.0 (1.0.1) to /Users/username/Library/Logs/DiagnosticReports/Link_2015-02-15-210338_MacBook-Pro-GS.crash
15/02/15 21:03:38,366 ReportCrash[8730]: Removing excessive log: file:///Users/username/Library/Logs/DiagnosticReports/Link_2015-02-15-195658_MacBook-GSPro.crash

**Comment by GnrGnr** [comment]
And the diagnostic report the from the last line:

Process:               Link [8714]
Path:                  /Users/USER/Downloads/LinkLiarDebug.app/Contents/MacOS/Link
Identifier:            com.funkensturm.Link
Version:               1.0 (1.0.1)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Link [8714]
User ID:               501

Date/Time:             2015-02-15 21:03:37.763 +0100
OS Version:            Mac OS X 10.10.2 (14C109)
Report Version:        11
Anonymous UUID:        E07342E5-8DC3-2AAB-E541-EA2D250E64D4

Sleep/Wake UUID:       B27A8E98-9E55-4567-9E36-8E2F7063E7B1

Time Awake Since Boot: 250000 seconds
Time Since Wake:       61 seconds

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Application Specific Information:
*** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[NSArray objectsAtIndexes:]: index 2 in index set beyond bounds [0 .. 1]'
terminating with uncaught exception of type NSException
abort() called

Application Specific Backtrace 1:
0   CoreFoundation                      0x00007fff8e85d66c __exceptionPreprocess + 172
1   libobjc.A.dylib                     0x00007fff934df76e objc_exception_throw + 43
2   CoreFoundation                      0x00007fff8e792933 -[NSArray objectsAtIndexes:] + 995
3   Link                                0x0000000102f73aec Link + 105196
4   Link                                0x0000000102f6f7e1 Link + 88033
5   Link                                0x0000000102f6a4a1 Link + 66721
6   Link                                0x0000000102f6a417 Link + 66583
7   Link                                0x0000000102f6a1f4 Link + 66036
8   Link                                0x0000000102f6a088 Link + 65672
9   Link                                0x0000000102f76ab3 Link + 117427
10  CoreFoundation                      0x00007fff8e76b85f -[NSSet makeObjectsPerformSelector:] + 223
11  AppKit                              0x00007fff8f4035cd -[NSIBObjectData nibInstantiateWithOwner:options:topLevelObjects:] + 1216
12  AppKit                              0x00007fff8ef13605 loadNib + 384
13  AppKit                              0x00007fff8f4845ab +[NSBundle(NSNibLoading) _loadNibFile:nameTable:options:withZone:ownerBundle:] + 313
14  AppKit                              0x00007fff8ef12cd5 -[NSBundle(NSNibLoading) loadNibNamed:owner:topLevelObjects:] + 201
15  AppKit                              0x00007fff8ef12aa1 +[NSBundle(NSNibLoading) loadNibNamed:owner:] + 344
16  AppKit                              0x00007fff8ef0e549 NSApplicationMain + 605
17  libdyld.dylib                       0x00007fff8c1725c9 start + 1

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff948d3286 __pthread_kill + 10
1   libsystem_c.dylib             	0x00007fff8e1d8b53 abort + 129
2   libc++abi.dylib               	0x00007fff8e145a21 abort_message + 257
3   libc++abi.dylib               	0x00007fff8e16d9d1 default_terminate_handler() + 267
4   libobjc.A.dylib               	0x00007fff934e37d6 _objc_terminate() + 103
5   libc++abi.dylib               	0x00007fff8e16b0a1 std::__terminate(void (*)()) + 8
6   libc++abi.dylib               	0x00007fff8e16ab30 __cxa_throw + 121
7   libobjc.A.dylib               	0x00007fff934df898 objc_exception_throw + 341
8   com.apple.CoreFoundation      	0x00007fff8e792933 -[NSArray objectsAtIndexes:] + 995
9   com.funkensturm.Link          	0x0000000102f73aec 0x102f5a000 + 105196
10  com.funkensturm.Link          	0x0000000102f6f7e1 0x102f5a000 + 88033
11  com.funkensturm.Link          	0x0000000102f6a4a1 0x102f5a000 + 66721
12  com.funkensturm.Link          	0x0000000102f6a417 0x102f5a000 + 66583
13  com.funkensturm.Link          	0x0000000102f6a1f4 0x102f5a000 + 66036
14  com.funkensturm.Link          	0x0000000102f6a088 0x102f5a000 + 65672
15  com.funkensturm.Link          	0x0000000102f76ab3 0x102f5a000 + 117427
16  com.apple.CoreFoundation      	0x00007fff8e76b85f -[NSSet makeObjectsPerformSelector:] + 223
17  com.apple.AppKit              	0x00007fff8f4035cd -[NSIBObjectData nibInstantiateWithOwner:options:topLevelObjects:] + 1216
18  com.apple.AppKit              	0x00007fff8ef13605 loadNib + 384
19  com.apple.AppKit              	0x00007fff8f4845ab +[NSBundle(NSNibLoading) _loadNibFile:nameTable:options:withZone:ownerBundle:] + 313
20  com.apple.AppKit              	0x00007fff8ef12cd5 -[NSBundle(NSNibLoading) loadNibNamed:owner:topLevelObjects:] + 201
21  com.apple.AppKit              	0x00007fff8ef12aa1 +[NSBundle(NSNibLoading) loadNibNamed:owner:] + 344
22  com.apple.AppKit              	0x00007fff8ef0e549 NSApplicationMain + 605
23  libdyld.dylib                 	0x00007fff8c1725c9 start + 1

Thread 1:: Dispatch queue: com.apple.libdispatch-manager
0   libsystem_kernel.dylib        	0x00007fff948d4232 kevent64 + 10
1   libdispatch.dylib             	0x00007fff947e6a6a _dispatch_mgr_thread + 52

Thread 2:
0   libsystem_kernel.dylib        	0x00007fff948d394a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff8e17240d start_wqthread + 13

Thread 3:
0   libsystem_kernel.dylib        	0x00007fff948d394a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff8e17240d start_wqthread + 13

Thread 4:: Dispatch queue: com.apple.NSXPCConnection.m-user.com.funkensturm.LinkHelper
0   libsystem_kernel.dylib        	0x00007fff948ce51a semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff947eac55 _dispatch_semaphore_wait_slow + 213
2   libxpc.dylib                  	0x00007fff948e47d1 xpc_connection_send_message_with_reply_sync + 200
3   com.apple.security            	0x00007fff92681ed7 AuthorizationCreate + 151
4   com.funkensturm.Link          	0x0000000102f7331b 0x102f5a000 + 103195
5   com.funkensturm.Link          	0x0000000102f72c99 0x102f5a000 + 101529
6   com.funkensturm.Link          	0x0000000102f71699 0x102f5a000 + 95897
7   com.apple.Foundation          	0x00007fff8a4c66f3 __NSXPCCONNECTION_IS_CALLING_OUT_TO_ERROR_BLOCK__ + 7
8   com.apple.Foundation          	0x00007fff8a4c7610 __97-[NSXPCConnection _sendInvocation:proxyNumber:remoteInterface:withErrorHandler:timeout:userInfo:]_block_invoke311 + 877
9   libxpc.dylib                  	0x00007fff948e4bfc _xpc_connection_reply_callout + 47
10  libxpc.dylib                  	0x00007fff948e4b88 _xpc_connection_call_reply + 36
11  libdispatch.dylib             	0x00007fff947e3c13 _dispatch_client_callout + 8
12  libdispatch.dylib             	0x00007fff947e7365 _dispatch_queue_drain + 1100
13  libdispatch.dylib             	0x00007fff947e8ecc _dispatch_queue_invoke + 202
14  libdispatch.dylib             	0x00007fff947e66b7 _dispatch_root_queue_drain + 463
15  libdispatch.dylib             	0x00007fff947f4fe4 _dispatch_worker_thread3 + 91
16  libsystem_pthread.dylib       	0x00007fff8e174637 _pthread_wqthread + 729
17  libsystem_pthread.dylib       	0x00007fff8e17240d start_wqthread + 13

Thread 5:
0   libsystem_kernel.dylib        	0x00007fff948d394a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff8e17240d start_wqthread + 13

Thread 6:
0   libsystem_kernel.dylib        	0x00007fff948d394a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff8e17240d start_wqthread + 13

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000006  rcx: 0x00007fff5cca3998  rdx: 0x0000000000000000
  rdi: 0x000000000000130f  rsi: 0x0000000000000006  rbp: 0x00007fff5cca39c0  rsp: 0x00007fff5cca3998
   r8: 0x0000000000000000   r9: 0x00007fff8e202d70  r10: 0x0000000008000000  r11: 0x0000000000000206
  r12: 0x00007fff5cca3b20  r13: 0x0000000000000002  r14: 0x00007fff79dea300  r15: 0x00007fff5cca3a00
  rip: 0x00007fff948d3286  rfl: 0x0000000000000206  cr2: 0x00007fff79df0fd8
  
Logical CPU:     0
Error Code:      0x02000148
Trap Number:     133


Binary Images:
       0x102f5a000 -        0x102f7efff +com.funkensturm.Link (1.0 - 1.0.1) <1E09391C-4768-3759-8229-A4B5A1520550> /Users/USER/Downloads/LinkLiarDebug.app/Contents/MacOS/Link
    0x7fff6cabf000 -     0x7fff6caf5837  dyld (353.2.1) <65DCCB06-339C-3E25-9702-600A28291D0E> /usr/lib/dyld
    0x7fff89a36000 -     0x7fff89a38fff  com.apple.CoreDuetDebugLogging (1.0 - 1) <9A6E5710-EA99-366E-BF40-9A65EC1B46A1> /System/Library/PrivateFrameworks/CoreDuetDebugLogging.framework/Versions/A/CoreDuetDebugLogging
    0x7fff89a39000 -     0x7fff89a3cfff  com.apple.IOSurface (97 - 97) <D4B4D2B2-7B16-3174-9EA6-55E0A10B452D> /System/Library/Frameworks/IOSurface.framework/Versions/A/IOSurface
    0x7fff89ad4000 -     0x7fff89c62fff  libBLAS.dylib (1128) <497912C1-A98E-3281-BED7-E9C751552F61> /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
    0x7fff89c63000 -     0x7fff89c8bffb  libRIP.A.dylib (775.16) <7711F7A7-1813-3024-AE42-75CA7C5422B7> /System/Library/Frameworks/CoreGraphics.framework/Versions/A/Resources/libRIP.A.dylib
    0x7fff89ce9000 -     0x7fff89cebff7  libsystem_sandbox.dylib (358.1.1) <95312E09-DA28-324A-A084-F3E574D0210E> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff89ef1000 -     0x7fff89ef3fff  com.apple.loginsupport (1.0 - 1) <21DBC18C-F260-39FC-B52F-04A5AA84523A> /System/Library/PrivateFrameworks/login.framework/Versions/A/Frameworks/loginsupport.framework/Versions/A/loginsupport
    0x7fff8a0c8000 -     0x7fff8a15cfff  com.apple.ink.framework (10.9 - 213) <8E029630-1530-3734-A446-13353F0E7AC5> /System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/Ink.framework/Versions/A/Ink
    0x7fff8a1d6000 -     0x7fff8a24afff  com.apple.ApplicationServices.ATS (360 - 375) <2824D38D-460D-353C-9D18-499B4BEEABB7> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/ATS
    0x7fff8a24b000 -     0x7fff8a579fff  com.apple.Foundation (6.9 - 1152.14) <E3746EDD-DFB1-3ECB-88ED-A91AC0EF3AAA> /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
    0x7fff8a57a000 -     0x7fff8add1ffb  com.apple.CoreGraphics (1.600.0 - 775.16) <864C1845-C41E-314C-A3B4-438DC39E5FBC> /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics
    0x7fff8adde000 -     0x7fff8ade7ff7  libsystem_notify.dylib (133.1.1) <61147800-F320-3DAA-850C-BADF33855F29> /usr/lib/system/libsystem_notify.dylib
    0x7fff8ade8000 -     0x7fff8ade9fff  libsystem_secinit.dylib (18) <581DAD0F-6B63-3A48-B63B-917AF799ABAA> /usr/lib/system/libsystem_secinit.dylib
    0x7fff8adea000 -     0x7fff8adeefff  libcache.dylib (69) <45E9A2E7-99C4-36B2-BEE3-0C4E11614AD1> /usr/lib/system/libcache.dylib
    0x7fff8b0de000 -     0x7fff8b116fff  com.apple.RemoteViewServices (2.0 - 99) <C9A62691-B0D9-34B7-B71C-A48B5F4DC553> /System/Library/PrivateFrameworks/RemoteViewServices.framework/Versions/A/RemoteViewServices
    0x7fff8b24a000 -     0x7fff8b4f6fff  com.apple.GeoServices (1.0 - 982.4.10) <8A7FE04A-2785-30E7-A6E2-DC15D170DAF5> /System/Library/PrivateFrameworks/GeoServices.framework/Versions/A/GeoServices
    0x7fff8b552000 -     0x7fff8b7cdff7  com.apple.CoreData (111 - 526.1) <DC4F037B-B7F4-381A-B939-4414489D76BF> /System/Library/Frameworks/CoreData.framework/Versions/A/CoreData
    0x7fff8b860000 -     0x7fff8b883fff  com.apple.Sharing (328.3.2 - 328.3.2) <F555679F-1CD1-3EB2-8E01-FCB80EF07330> /System/Library/PrivateFrameworks/Sharing.framework/Versions/A/Sharing
    0x7fff8b8da000 -     0x7fff8b8f4ff7  liblzma.5.dylib (7) <1D03E875-A7C0-3028-814C-3C27F7B7C079> /usr/lib/liblzma.5.dylib
    0x7fff8b8f5000 -     0x7fff8b90ffff  com.apple.AppleVPAFramework (1.2.10 - 1.2.10) <DC3D5A44-AB1E-32A9-9D22-FC922B52346A> /System/Library/PrivateFrameworks/AppleVPA.framework/Versions/A/AppleVPA
    0x7fff8b920000 -     0x7fff8b98effb  com.apple.Heimdal (4.0 - 2.0) <3E5DA653-A343-3257-ADE1-BA879BAE280F> /System/Library/PrivateFrameworks/Heimdal.framework/Versions/A/Heimdal
    0x7fff8b9fd000 -     0x7fff8ba01fff  com.apple.TCC (1.0 - 1) <61F36A72-B983-3A2D-9D37-A2F194D31E7D> /System/Library/PrivateFrameworks/TCC.framework/Versions/A/TCC
    0x7fff8ba7b000 -     0x7fff8bae7fff  com.apple.framework.CoreWLAN (5.0 - 500.35.2) <37551DDD-C07C-31EB-923A-9721F03D7E29> /System/Library/Frameworks/CoreWLAN.framework/Versions/A/CoreWLAN
    0x7fff8bb53000 -     0x7fff8bb57fff  libCoreVMClient.dylib (79) <FC4E08E3-749E-32FF-B5E9-211F29864831> /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libCoreVMClient.dylib
    0x7fff8bb58000 -     0x7fff8bb58fff  libOpenScriptingUtil.dylib (162) <EFD79173-A9DA-3AE6-BE15-3948938204A6> /usr/lib/libOpenScriptingUtil.dylib
    0x7fff8bbe7000 -     0x7fff8bc12fff  com.apple.DictionaryServices (1.2 - 229) <6789EC43-CADA-394D-8FE8-FC3A2DD136B9> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/DictionaryServices.framework/Versions/A/DictionaryServices
    0x7fff8bc13000 -     0x7fff8bc59ff7  libauto.dylib (186) <A260789B-D4D8-316A-9490-254767B8A5F1> /usr/lib/libauto.dylib
    0x7fff8bc5a000 -     0x7fff8bcccff7  com.apple.framework.IOKit (2.0.2 - 1050.10.8) <FDFB1FBE-6A0E-3D63-828C-CD53500FCB0F> /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
    0x7fff8bccd000 -     0x7fff8bcddff7  libbsm.0.dylib (34) <A3A2E56C-2B65-37C7-B43A-A1F926E1A0BB> /usr/lib/libbsm.0.dylib
    0x7fff8bd44000 -     0x7fff8bd51fff  com.apple.SpeechRecognitionCore (2.0.32 - 2.0.32) <87F0C88D-502D-3217-8B4A-8388288568BA> /System/Library/PrivateFrameworks/SpeechRecognitionCore.framework/Versions/A/SpeechRecognitionCore
    0x7fff8bdb4000 -     0x7fff8bdbcffb  com.apple.CoreServices.FSEvents (1210 - 1210) <782A9C69-7A45-31A7-8960-D08A36CBD0A7> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/FSEvents.framework/Versions/A/FSEvents
    0x7fff8bdbd000 -     0x7fff8bdc8ff7  com.apple.speech.synthesis.framework (5.3.3 - 5.3.3) <7DF3C68C-B219-3E13-AE72-24B8606A1560> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/SpeechSynthesis.framework/Versions/A/SpeechSynthesis
    0x7fff8c16f000 -     0x7fff8c172ff7  libdyld.dylib (353.2.1) <4E33E416-F1D8-3598-B8CC-6863E2ECD0E6> /usr/lib/system/libdyld.dylib
    0x7fff8c173000 -     0x7fff8c1f0fff  com.apple.CoreServices.OSServices (640.3 - 640.3) <84A91B00-0ED4-350C-B30A-AEAE437AE02A> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/OSServices.framework/Versions/A/OSServices
    0x7fff8c24d000 -     0x7fff8c254fff  libCGCMS.A.dylib (775.16) <8A173E74-7123-35F1-B160-853528C144ED> /System/Library/Frameworks/CoreGraphics.framework/Versions/A/Resources/libCGCMS.A.dylib
    0x7fff8c25e000 -     0x7fff8c562ffb  com.apple.HIToolbox (2.1.1 - 757.3) <D827FC03-5668-3AA4-AF0E-46EEF7358EEA> /System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/HIToolbox
    0x7fff8c563000 -     0x7fff8c579ff7  libsystem_asl.dylib (267) <F153AC5B-0542-356E-88C8-20A62CA704E2> /usr/lib/system/libsystem_asl.dylib
    0x7fff8c57f000 -     0x7fff8c5adfff  com.apple.CoreServicesInternal (221.2.2 - 221.2.2) <16F7A7F1-CF1D-35AD-A91F-690A814048DF> /System/Library/PrivateFrameworks/CoreServicesInternal.framework/Versions/A/CoreServicesInternal
    0x7fff8c5c5000 -     0x7fff8c5c5fff  com.apple.ApplicationServices (48 - 48) <5BF7910B-C328-3BF8-BA4F-CE52B574CE01> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices
    0x7fff8c818000 -     0x7fff8c820fff  libsystem_dnssd.dylib (561.1.1) <62B70ECA-E40D-3C63-896E-7F00EC386DDB> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff8c901000 -     0x7fff8c979ff7  com.apple.SystemConfiguration (1.14 - 1.14) <E0495F7D-5624-3EF7-B7E5-DA0EE708B6E4> /System/Library/Frameworks/SystemConfiguration.framework/Versions/A/SystemConfiguration
    0x7fff8c981000 -     0x7fff8c99cff7  libCRFSuite.dylib (34) <D64842BE-7BD4-3D0C-9842-1D202F7C2A51> /usr/lib/libCRFSuite.dylib
    0x7fff8c99d000 -     0x7fff8c9a2ff7  libmacho.dylib (862) <126CA2ED-DE91-308F-8881-B9DAEC3C63B6> /usr/lib/system/libmacho.dylib
    0x7fff8c9a3000 -     0x7fff8ca0affb  com.apple.datadetectorscore (6.0 - 396.1.1) <80379385-A4EC-3F9B-AFED-9B1DF781943D> /System/Library/PrivateFrameworks/DataDetectorsCore.framework/Versions/A/DataDetectorsCore
    0x7fff8ca0b000 -     0x7fff8ca26ff7  com.apple.aps.framework (4.0 - 4.0) <F3C3C246-101E-3E81-9608-D2D6E9352532> /System/Library/PrivateFrameworks/ApplePushService.framework/Versions/A/ApplePushService
    0x7fff8dbfb000 -     0x7fff8dc04ff3  com.apple.CommonAuth (4.0 - 2.0) <BA9F5A09-D200-3D18-9F4A-20C789291A30> /System/Library/PrivateFrameworks/CommonAuth.framework/Versions/A/CommonAuth
    0x7fff8dc05000 -     0x7fff8dc0dfff  libsystem_platform.dylib (63) <64E34079-D712-3D66-9CE2-418624A5C040> /usr/lib/system/libsystem_platform.dylib
    0x7fff8dc0e000 -     0x7fff8dc1bff7  libbz2.1.0.dylib (36) <2DF83FBC-5C08-39E1-94F5-C28653791B5F> /usr/lib/libbz2.1.0.dylib
    0x7fff8dc1c000 -     0x7fff8dc33ff7  libLinearAlgebra.dylib (1128) <E78CCBAA-A999-3B65-8EC9-06DB15E67C37> /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLinearAlgebra.dylib
    0x7fff8dc49000 -     0x7fff8dc84fff  com.apple.QD (301 - 301) <C4D2AD03-B839-350A-AAF0-B4A08F8BED77> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/QD.framework/Versions/A/QD
    0x7fff8dd14000 -     0x7fff8e144fff  com.apple.vision.FaceCore (3.1.6 - 3.1.6) <C3B823AA-C261-37D3-B4AC-C59CE91C8241> /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/FaceCore
    0x7fff8e145000 -     0x7fff8e170fff  libc++abi.dylib (125) <88A22A0F-87C6-3002-BFBA-AC0F2808B8B9> /usr/lib/libc++abi.dylib
    0x7fff8e171000 -     0x7fff8e17afff  libsystem_pthread.dylib (105.10.1) <3103AA7F-3BAE-3673-9649-47FFD7E15C97> /usr/lib/system/libsystem_pthread.dylib
    0x7fff8e17b000 -     0x7fff8e207ff7  libsystem_c.dylib (1044.10.1) <199ED5EB-77A1-3D43-AA51-81779CE0A742> /usr/lib/system/libsystem_c.dylib
    0x7fff8e208000 -     0x7fff8e320ffb  com.apple.CoreText (352.0 - 454.3) <B3B8C775-14FA-38F3-9CD5-830422AE9C49> /System/Library/Frameworks/CoreText.framework/Versions/A/CoreText
    0x7fff8e321000 -     0x7fff8e414ff7  libJP2.dylib (1232) <10B78725-0B8A-3D87-B2E3-8FEED0C07F21> /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libJP2.dylib
    0x7fff8e651000 -     0x7fff8e6f3ff7  com.apple.Bluetooth (4.3.2 - 4.3.2f6) <95676652-21AB-3FFA-B53D-EBC8BF4E913E> /System/Library/Frameworks/IOBluetooth.framework/Versions/A/IOBluetooth
    0x7fff8e6fb000 -     0x7fff8ea91fff  com.apple.CoreFoundation (6.9 - 1152) <CBD1591C-405E-376E-87E9-B264610EBF49> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff8ea92000 -     0x7fff8eadbff3  com.apple.HIServices (1.22 - 520.12) <8EAC82AB-6A7D-3606-AF6F-60A9410D1278> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/HIServices.framework/Versions/A/HIServices
    0x7fff8eadc000 -     0x7fff8eae2fff  com.apple.speech.recognition.framework (5.0.9 - 5.0.9) <BB2D573F-0A01-379F-A2BA-3C454EDCB111> /System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/SpeechRecognition.framework/Versions/A/SpeechRecognition
    0x7fff8eae3000 -     0x7fff8eafdff3  com.apple.Ubiquity (1.3 - 313) <DF56A657-CC6E-3BE2-86A0-71F07127724C> /System/Library/PrivateFrameworks/Ubiquity.framework/Versions/A/Ubiquity
    0x7fff8eb30000 -     0x7fff8eb7dff3  com.apple.print.framework.PrintCore (10.0 - 451) <3CA58254-D14F-3913-9DFB-CAC499570CC7> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/PrintCore.framework/Versions/A/PrintCore
    0x7fff8ebfa000 -     0x7fff8ec83fff  com.apple.CoreSymbolication (3.1 - 57020) <FDF8F348-164D-38F9-90EB-F42585DD2C77> /System/Library/PrivateFrameworks/CoreSymbolication.framework/Versions/A/CoreSymbolication
    0x7fff8ecb2000 -     0x7fff8ed50fff  com.apple.Metadata (10.7.0 - 917.1) <46BE997C-B1F4-3BED-9332-FAC87297C87A> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/Metadata.framework/Versions/A/Metadata
    0x7fff8ed51000 -     0x7fff8ed55ff7  libGIF.dylib (1232) <061D5354-FE4F-3C7E-B563-99DC0198062D> /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libGIF.dylib
    0x7fff8ed56000 -     0x7fff8ed5bfff  com.apple.DiskArbitration (2.6 - 2.6) <0DFF4D9B-2AC3-3B82-B5C5-30F4EFBD2DB9> /System/Library/Frameworks/DiskArbitration.framework/Versions/A/DiskArbitration
    0x7fff8ed5c000 -     0x7fff8ed89fff  com.apple.Accounts (113 - 113) <990F0F61-6AC5-3076-932E-02A9A7F75AC4> /System/Library/Frameworks/Accounts.framework/Versions/A/Accounts
    0x7fff8ed8a000 -     0x7fff8ee7efff  libFontParser.dylib (134.1) <EA8452DB-9221-3608-95BF-496F58106313> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/Resources/libFontParser.dylib
    0x7fff8ee7f000 -     0x7fff8eef5fe7  libcorecrypto.dylib (233.1.2) <E1789801-3985-3949-B736-6B3378873301> /usr/lib/system/libcorecrypto.dylib
    0x7fff8ef0c000 -     0x7fff8fa56ff7  com.apple.AppKit (6.9 - 1344.72) <44EF7DEB-3072-3515-9F34-2857D557E828> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
    0x7fff8fa57000 -     0x7fff8fa59fff  libRadiance.dylib (1232) <E670DDEF-60F8-3AEB-B6A2-B20A1340634C> /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libRadiance.dylib
    0x7fff8fa67000 -     0x7fff8fa67fff  com.apple.Accelerate.vecLib (3.10 - vecLib 3.10) <B92888D0-ED3F-3430-8F3A-6E56FD16C5F1> /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/vecLib
    0x7fff902f4000 -     0x7fff90436fff  libsqlite3.dylib (168) <7B580EB9-9260-35FE-AE2F-276A2C242BAB> /usr/lib/libsqlite3.dylib
    0x7fff90437000 -     0x7fff90448ff7  libsystem_coretls.dylib (35.1.2) <BC691CD1-17B6-39A5-BD02-AF973695FD1D> /usr/lib/system/libsystem_coretls.dylib
    0x7fff90452000 -     0x7fff90493fff  libGLU.dylib (11.1.1) <E9ADAD30-0133-320D-A60E-D1A7F91A7795> /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGLU.dylib
    0x7fff9072b000 -     0x7fff90753fff  libsystem_info.dylib (459) <B85A85D5-8530-3A93-B0C3-4DEC41F79478> /usr/lib/system/libsystem_info.dylib
    0x7fff909d1000 -     0x7fff909d3fff  libsystem_configuration.dylib (699.1.5) <5E14864E-089A-3D84-85A4-980B776427A8> /usr/lib/system/libsystem_configuration.dylib
    0x7fff90a1c000 -     0x7fff90b87ff7  com.apple.audio.toolbox.AudioToolbox (1.12 - 1.12) <5C6DBEB4-F2EA-3262-B9FC-AFB89404C1DA> /System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox
    0x7fff918a9000 -     0x7fff918a9fff  com.apple.Cocoa (6.8 - 21) <EAC0EA1E-3C62-3B28-A941-5D8B1E085FF8> /System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa
    0x7fff918aa000 -     0x7fff91905fff  libTIFF.dylib (1232) <29A5C7F7-D50B-35B3-8FA2-A55A47E497A6> /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libTIFF.dylib
    0x7fff9190e000 -     0x7fff919addf7  com.apple.AppleJPEG (1.0 - 1) <9BB3D7DF-630A-3E1C-A124-12D6C4D0DE70> /System/Library/PrivateFrameworks/AppleJPEG.framework/Versions/A/AppleJPEG
    0x7fff919d7000 -     0x7fff91af9ff7  com.apple.LaunchServices (644.12.4 - 644.12.4) <59E909E8-ED4A-33EA-B85D-D409BADDF854> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/LaunchServices
    0x7fff91afa000 -     0x7fff91afbfff  libDiagnosticMessagesClient.dylib (100) <2EE8E436-5CDC-34C5-9959-5BA218D507FB> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fff91bf0000 -     0x7fff91bf4fff  libpam.2.dylib (20) <E805398D-9A92-31F8-8005-8DC188BD8B6E> /usr/lib/libpam.2.dylib
    0x7fff91c02000 -     0x7fff91cf4ff7  libiconv.2.dylib (42) <2A06D02F-8B76-3864-8D96-64EF5B40BC6C> /usr/lib/libiconv.2.dylib
    0x7fff91cf5000 -     0x7fff91cffff7  com.apple.CrashReporterSupport (10.10 - 629) <4BCAA6B5-EC7F-365F-9D3F-BC483B7E956C> /System/Library/PrivateFrameworks/CrashReporterSupport.framework/Versions/A/CrashReporterSupport
    0x7fff91d10000 -     0x7fff92043fff  libmecabra.dylib (666.2) <F757CABA-3EDB-3ABA-A378-A7C574EA233B> /usr/lib/libmecabra.dylib
    0x7fff92044000 -     0x7fff92045ff7  libsystem_blocks.dylib (65) <9615D10A-FCA7-3BE4-AA1A-1B195DACE1A1> /usr/lib/system/libsystem_blocks.dylib
    0x7fff92123000 -     0x7fff9212bffb  libcopyfile.dylib (118.1.2) <0C68D3A6-ACDD-3EF3-991A-CC82C32AB836> /usr/lib/system/libcopyfile.dylib
    0x7fff92375000 -     0x7fff92483fff  com.apple.desktopservices (1.9.2 - 1.9.2) <8670FD3B-8A5B-3D84-B21E-DF21140545A2> /System/Library/PrivateFrameworks/DesktopServicesPriv.framework/Versions/A/DesktopServicesPriv
    0x7fff92503000 -     0x7fff92549ffb  libFontRegistry.dylib (134) <01B8034A-45FD-3360-A347-A1896F591363> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/Resources/libFontRegistry.dylib
    0x7fff92559000 -     0x7fff92566ff7  libxar.1.dylib (254) <CE10EFED-3066-3749-838A-6A15AC0DBCB6> /usr/lib/libxar.1.dylib
    0x7fff925cd000 -     0x7fff92835ff3  com.apple.security (7.0 - 57031.10.10) <79C37E73-271B-3BEF-A96E-CDB83FF12CF0> /System/Library/Frameworks/Security.framework/Versions/A/Security
    0x7fff9288c000 -     0x7fff928e6ff7  com.apple.LanguageModeling (1.0 - 1) <ACA93FE0-A0E3-333E-AE3C-8EB7DE5F362F> /System/Library/PrivateFrameworks/LanguageModeling.framework/Versions/A/LanguageModeling
    0x7fff9292a000 -     0x7fff92938ff7  com.apple.opengl (11.1.1 - 11.1.1) <F79F5FFF-372E-329E-81FB-EE9BD6A2A7A7> /System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL
    0x7fff92939000 -     0x7fff92946fff  com.apple.ProtocolBuffer (1 - 225.1) <2D502FBB-D2A0-3937-A5C5-385FA65B3874> /System/Library/PrivateFrameworks/ProtocolBuffer.framework/Versions/A/ProtocolBuffer
    0x7fff92947000 -     0x7fff92c2effb  com.apple.CoreServices.CarbonCore (1108.2 - 1108.2) <FD87F83F-301A-3BD6-8262-5692FC1B4457> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/CarbonCore.framework/Versions/A/CarbonCore
    0x7fff92c2f000 -     0x7fff92d57ff7  com.apple.coreui (2.1 - 305.6.1) <B56EC212-73C1-326F-B78C-EB856386296E> /System/Library/PrivateFrameworks/CoreUI.framework/Versions/A/CoreUI
    0x7fff92d58000 -     0x7fff92d58fff  com.apple.audio.units.AudioUnit (1.12 - 1.12) <76EF1C9D-DEA4-3E55-A134-4099B2FD2CF2> /System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit
    0x7fff92d62000 -     0x7fff92d87ff7  libPng.dylib (1232) <6E72AE55-AFB0-3FC4-80B2-EBC3353436B7> /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib
    0x7fff9308a000 -     0x7fff93091fff  com.apple.NetFS (6.0 - 4.0) <1581D25F-CC07-39B0-90E8-5D4F3CF84EBA> /System/Library/Frameworks/NetFS.framework/Versions/A/NetFS
    0x7fff930e2000 -     0x7fff93164fff  com.apple.PerformanceAnalysis (1.0 - 1) <94F08B1A-F6AF-38D5-BE92-4FED34742966> /System/Library/PrivateFrameworks/PerformanceAnalysis.framework/Versions/A/PerformanceAnalysis
    0x7fff934d1000 -     0x7fff936cb46f  libobjc.A.dylib (647) <759E155D-BC42-3D4E-869B-6F57D477177C> /usr/lib/libobjc.A.dylib
    0x7fff936cc000 -     0x7fff936e5ff7  com.apple.CFOpenDirectory (10.10 - 187) <0F9747EF-12A3-3694-984D-0B8352CA6C0F> /System/Library/Frameworks/OpenDirectory.framework/Versions/A/Frameworks/CFOpenDirectory.framework/Versions/A/CFOpenDirectory
    0x7fff937ab000 -     0x7fff937b6fff  libGL.dylib (11.1.1) <1F0EB9FB-4B0F-349B-80DD-93FD3F45B9C7> /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib
    0x7fff93cfc000 -     0x7fff93dbfff7  libvMisc.dylib (516) <A84F3A3B-D349-3FBC-B5A6-E0F572734073> /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libvMisc.dylib
    0x7fff93dc0000 -     0x7fff93e11ff7  com.apple.audio.CoreAudio (4.3.0 - 4.3.0) <AF72B06E-C6C1-3FAE-8B47-AF461CAE0E22> /System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio
    0x7fff93e12000 -     0x7fff93e2cff7  com.apple.Kerberos (3.0 - 1) <7760E0C2-A222-3709-B2A6-B692D900CEB1> /System/Library/Frameworks/Kerberos.framework/Versions/A/Kerberos
    0x7fff93ec6000 -     0x7fff93f00ffb  com.apple.DebugSymbols (115 - 115) <6F03761D-7C3A-3C80-8031-AA1C1AD7C706> /System/Library/PrivateFrameworks/DebugSymbols.framework/Versions/A/DebugSymbols
    0x7fff93f01000 -     0x7fff93f22fff  com.apple.framework.Apple80211 (10.1 - 1010.64) <A7378C4B-FFD3-35B9-93E8-0534A2A7B51F> /System/Library/PrivateFrameworks/Apple80211.framework/Versions/A/Apple80211
    0x7fff93fdf000 -     0x7fff93fe1fff  com.apple.OAuth (25 - 25) <EE765AF0-2BB6-3689-9EAA-689BF1F02A0D> /System/Library/PrivateFrameworks/OAuth.framework/Versions/A/OAuth
    0x7fff94656000 -     0x7fff94716fff  com.apple.backup.framework (1.6.2 - 1.6.2) <63E8CA47-B7B8-3A63-B505-D1622CE52527> /System/Library/PrivateFrameworks/Backup.framework/Versions/A/Backup
    0x7fff9472f000 -     0x7fff947c0ff7  com.apple.cloudkit.CloudKit (259.2.5 - 259.2.5) <241EB647-C917-32F7-956A-6E505827048C> /System/Library/Frameworks/CloudKit.framework/Versions/A/CloudKit
    0x7fff947e2000 -     0x7fff9480cff7  libdispatch.dylib (442.1.4) <502CF32B-669B-3709-8862-08188225E4F0> /usr/lib/system/libdispatch.dylib
    0x7fff94841000 -     0x7fff9484cff7  libkxld.dylib (2782.10.72) <68E07A32-28F5-3FBB-9D74-00B4F53C2FD4> /usr/lib/system/libkxld.dylib
    0x7fff9484d000 -     0x7fff94855ff7  com.apple.AppleSRP (5.0 - 1) <01EC5144-D09A-3D6A-AE35-F6D48585F154> /System/Library/PrivateFrameworks/AppleSRP.framework/Versions/A/AppleSRP
    0x7fff94856000 -     0x7fff94883fff  com.apple.CoreVideo (1.8 - 145.1) <18DB07E0-B927-3260-A234-636F298D1917> /System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo
    0x7fff948bd000 -     0x7fff948dafff  libsystem_kernel.dylib (2782.10.72) <97CD7ACD-EA0C-3434-BEFC-FCD013D6BB73> /usr/lib/system/libsystem_kernel.dylib
    0x7fff948db000 -     0x7fff94903fff  libxpc.dylib (559.10.3) <876216DC-D5D3-381E-8AF9-49AE464E5107> /usr/lib/system/libxpc.dylib
    0x7fff94989000 -     0x7fff94994fff  libcommonCrypto.dylib (60061) <D381EBC6-69D8-31D3-8084-5A80A32CB748> /usr/lib/system/libcommonCrypto.dylib
    0x7fff949dd000 -     0x7fff949ddfff  com.apple.CoreServices (62 - 62) <9E4577CA-3FC3-300D-AB00-87ADBDDA2E37> /System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices
    0x7fff94a34000 -     0x7fff94a53fff  com.apple.CoreDuet (1.0 - 1) <36AA9FD5-2685-314D-B364-3FA4688D86BD> /System/Library/PrivateFrameworks/CoreDuet.framework/Versions/A/CoreDuet
    0x7fff94a54000 -     0x7fff94a94ff7  libGLImage.dylib (11.1.1) <3986BFA3-4F55-380F-B01D-91BA9785D70C> /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGLImage.dylib
    0x7fff9500e000 -     0x7fff95010ff7  libquarantine.dylib (76) <DC041627-2D92-361C-BABF-A869A5C72293> /usr/lib/system/libquarantine.dylib
    0x7fff95011000 -     0x7fff950f5fff  libcrypto.0.9.8.dylib (52.10.1) <2A2924DE-63FB-37F6-B102-84D69240675B> /usr/lib/libcrypto.0.9.8.dylib
    0x7fff95128000 -     0x7fff951bdff7  com.apple.ColorSync (4.9.0 - 4.9.0) <F06733BD-A10C-3DB3-B050-825351130392> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ColorSync.framework/Versions/A/ColorSync
    0x7fff951be000 -     0x7fff951eefff  libsystem_m.dylib (3086.1) <1E12AB45-6D96-36D0-A226-F24D9FB0D9D6> /usr/lib/system/libsystem_m.dylib
    0x7fff952ef000 -     0x7fff9535efff  com.apple.SearchKit (1.4.0 - 1.4.0) <BFD6D876-36BA-3A3B-9F15-3E2F7DE6E89D> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SearchKit.framework/Versions/A/SearchKit
    0x7fff9535f000 -     0x7fff9549cfff  com.apple.ImageIO.framework (3.3.0 - 1232) <D7AF3CD2-FAB2-3798-9C26-914886852DCD> /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
    0x7fff954a2000 -     0x7fff954a7ff7  libunwind.dylib (35.3) <BE7E51A0-B6EA-3A54-9CCA-9D88F683A6D6> /usr/lib/system/libunwind.dylib
    0x7fff954a8000 -     0x7fff954a8ff7  libkeymgr.dylib (28) <77845842-DE70-3CC5-BD01-C3D14227CED5> /usr/lib/system/libkeymgr.dylib
    0x7fff954a9000 -     0x7fff954abff7  libsystem_coreservices.dylib (9) <41B7C578-5A53-31C8-A96F-C73E030B0938> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff954ac000 -     0x7fff954d2ff7  com.apple.ChunkingLibrary (2.1 - 163.1) <3514F2A4-38BD-3849-9286-B3B991057742> /System/Library/PrivateFrameworks/ChunkingLibrary.framework/Versions/A/ChunkingLibrary
    0x7fff954d4000 -     0x7fff954d6fff  libCVMSPluginSupport.dylib (11.1.1) <DA0706C5-F02A-3F3D-8EBA-18C04313CA2C> /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libCVMSPluginSupport.dylib
    0x7fff9565b000 -     0x7fff95660ffb  libheimdal-asn1.dylib (398.10.1) <A7B6447A-6680-3625-83C3-993B58D5C43F> /usr/lib/libheimdal-asn1.dylib
    0x7fff95663000 -     0x7fff95663ff7  liblaunch.dylib (559.10.3) <DFCDEBDF-8247-3DC7-9879-E7E497DDA4B4> /usr/lib/system/liblaunch.dylib
    0x7fff95664000 -     0x7fff95756fff  libxml2.2.dylib (26) <B834E7C8-EC3E-3382-BC5A-DA38DC4D720C> /usr/lib/libxml2.2.dylib
    0x7fff95757000 -     0x7fff957a3ff7  libcups.2.dylib (408) <9CECCDE3-51D7-3028-830C-F58BD36E3317> /usr/lib/libcups.2.dylib
    0x7fff957a4000 -     0x7fff957a4ff7  libunc.dylib (29) <5676F7EA-C1DF-329F-B006-D2C3022B7D70> /usr/lib/system/libunc.dylib
    0x7fff957f4000 -     0x7fff95853ff3  com.apple.AE (681 - 681) <7F544183-A515-31A8-B45F-89A167F56216> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/AE.framework/Versions/A/AE
    0x7fff95906000 -     0x7fff9590cfff  libsystem_trace.dylib (72.1.3) <A9E6B7D8-C327-3742-AC54-86C94218B1DF> /usr/lib/system/libsystem_trace.dylib
    0x7fff9590d000 -     0x7fff95a24fe7  libvDSP.dylib (516) <DFEDB210-49D1-3803-88A2-C61DB6A45C3D> /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libvDSP.dylib
    0x7fff95a46000 -     0x7fff95a58ff7  com.apple.CoreDuetDaemonProtocol (1.0 - 1) <CE9FABB4-1C5D-3F9B-9BB8-5CC50C3E5E31> /System/Library/PrivateFrameworks/CoreDuetDaemonProtocol.framework/Versions/A/CoreDuetDaemonProtocol
    0x7fff95a59000 -     0x7fff95aadfff  libc++.1.dylib (120) <1B9530FD-989B-3174-BB1C-BDC159501710> /usr/lib/libc++.1.dylib
    0x7fff95aae000 -     0x7fff95ab3ff7  libsystem_stats.dylib (163.10.18) <9B8CCF24-DDDB-399A-9237-4BEC225D2E8C> /usr/lib/system/libsystem_stats.dylib
    0x7fff95ab4000 -     0x7fff95d83ff3  com.apple.CoreImage (10.0.33) <6E3DDA29-718B-3BDB-BFAF-F8C201BF93A4> /System/Library/Frameworks/QuartzCore.framework/Versions/A/Frameworks/CoreImage.framework/Versions/A/CoreImage
    0x7fff95d87000 -     0x7fff95d98fff  libcmph.dylib (1) <46EC3997-DB5E-38AE-BBBB-A035A54AD3C0> /usr/lib/libcmph.dylib
    0x7fff95f63000 -     0x7fff95f7ffff  com.apple.GenerationalStorage (2.0 - 209.11) <9FF8DD11-25FB-3047-A5BF-9415339B3EEC> /System/Library/PrivateFrameworks/GenerationalStorage.framework/Versions/A/GenerationalStorage
    0x7fff95f80000 -     0x7fff95fccff7  com.apple.corelocation (1486.17 - 1615.21.1) <B81BC475-E215-3491-A750-8B23F05ABF5B> /System/Library/Frameworks/CoreLocation.framework/Versions/A/CoreLocation
    0x7fff95fd3000 -     0x7fff96005ff3  com.apple.frameworks.CoreDaemon (1.3 - 1.3) <C6DB0A07-F8E4-3837-BCA9-225F460EDA81> /System/Library/PrivateFrameworks/CoreDaemon.framework/Versions/B/CoreDaemon
    0x7fff9614b000 -     0x7fff962fbff7  com.apple.QuartzCore (1.10 - 361.15) <72A78C43-30DF-3748-9015-4B28119DB27B> /System/Library/Frameworks/QuartzCore.framework/Versions/A/QuartzCore
    0x7fff963ba000 -     0x7fff96421ff7  com.apple.framework.CoreWiFi (3.0 - 300.4) <19269C1D-EB29-384A-83F3-7DDDEB7D9DAD> /System/Library/PrivateFrameworks/CoreWiFi.framework/Versions/A/CoreWiFi
    0x7fff9644a000 -     0x7fff9664dff3  com.apple.CFNetwork (720.2.4 - 720.2.4) <E550C671-930F-3B12-8798-23898473E179> /System/Library/Frameworks/CFNetwork.framework/Versions/A/CFNetwork
    0x7fff966aa000 -     0x7fff966b3fff  libGFXShared.dylib (11.1.1) <7AE7D152-597E-3B27-A52C-8DA76760B61C> /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGFXShared.dylib
    0x7fff966b4000 -     0x7fff967e4fff  com.apple.UIFoundation (1.0 - 1) <8E030D93-441C-3997-9CD2-55C8DFAC8B84> /System/Library/PrivateFrameworks/UIFoundation.framework/Versions/A/UIFoundation
    0x7fff96812000 -     0x7fff9683dff3  libarchive.2.dylib (30) <8CBB4416-EBE9-3574-8ADC-44655D245F39> /usr/lib/libarchive.2.dylib
    0x7fff9683e000 -     0x7fff9683ffff  libSystem.B.dylib (1213) <90B107BC-FF74-32CC-B1CF-4E02F544D957> /usr/lib/libSystem.B.dylib
    0x7fff9684d000 -     0x7fff9688dff7  com.apple.CloudDocs (1.0 - 280.6) <C1179CEF-E058-3E16-BF90-C059FE7CDE77> /System/Library/PrivateFrameworks/CloudDocs.framework/Versions/A/CloudDocs
    0x7fff9688e000 -     0x7fff968c6ffb  libsystem_network.dylib (411.1) <2EC3A005-473F-3C36-A665-F88B5BACC7F0> /usr/lib/system/libsystem_network.dylib
    0x7fff968c7000 -     0x7fff968e4ffb  libresolv.9.dylib (57) <26B38E61-298A-3C3A-82C1-3B5E98AD5E29> /usr/lib/libresolv.9.dylib
    0x7fff968e5000 -     0x7fff9690effb  libxslt.1.dylib (13) <AED1143F-B848-3E73-81ED-71356F25F084> /usr/lib/libxslt.1.dylib
    0x7fff9690f000 -     0x7fff96af4ff3  libicucore.A.dylib (531.31) <B08E00D5-13C6-3391-AB3A-8DE693D3B42E> /usr/lib/libicucore.A.dylib
    0x7fff96af5000 -     0x7fff96b1aff7  libJPEG.dylib (1232) <09466709-4742-3418-A0AC-116EF9714E2D> /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libJPEG.dylib
    0x7fff96b55000 -     0x7fff96b56fff  com.apple.TrustEvaluationAgent (2.0 - 25) <2D61A2C3-C83E-3A3F-8EC1-736DBEC250AB> /System/Library/PrivateFrameworks/TrustEvaluationAgent.framework/Versions/A/TrustEvaluationAgent
    0x7fff96b57000 -     0x7fff96b66fff  com.apple.LangAnalysis (1.7.0 - 1.7.0) <D1E527E4-C561-352F-9457-E8C50232793C> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/LangAnalysis.framework/Versions/A/LangAnalysis
    0x7fff96b67000 -     0x7fff96b83ff7  libsystem_malloc.dylib (53.1.1) <19BCC257-5717-3502-A71F-95D65AFA861B> /usr/lib/system/libsystem_malloc.dylib
    0x7fff96b84000 -     0x7fff96b90ff7  com.apple.OpenDirectory (10.10 - 187) <8B98ECCB-7EFA-3A58-BD2B-A0835D869B1A> /System/Library/Frameworks/OpenDirectory.framework/Versions/A/OpenDirectory
    0x7fff96b91000 -     0x7fff96ba2ff7  libz.1.dylib (55) <88C7C7DE-04B8-316F-8B74-ACD9F3DE1AA1> /usr/lib/libz.1.dylib
    0x7fff96bdb000 -     0x7fff96be5ff7  com.apple.NetAuth (5.0 - 5.0) <B9EC5425-D38D-308C-865F-207E0A98BAC7> /System/Library/PrivateFrameworks/NetAuth.framework/Versions/A/NetAuth
    0x7fff96be6000 -     0x7fff96be6fff  com.apple.Accelerate (1.10 - Accelerate 1.10) <F1B96A61-7E4B-31BD-A35B-BA7EF1F16EF4> /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate
    0x7fff97196000 -     0x7fff975a3ff7  libLAPACK.dylib (1128) <F9201AE7-B031-36DB-BCF8-971E994EF7C1> /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib
    0x7fff976e7000 -     0x7fff976faff7  com.apple.CoreBluetooth (1.0 - 1) <FA9B43B3-E183-3040-AE25-66EF9870CF35> /System/Library/Frameworks/CoreBluetooth.framework/Versions/A/CoreBluetooth
    0x7fff976fb000 -     0x7fff97b4efc7  com.apple.vImage (8.0 - 8.0) <33BE7B31-72DB-3364-B37E-C322A32748C5> /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vImage.framework/Versions/A/vImage
    0x7fff97c73000 -     0x7fff97cc2ff7  com.apple.opencl (2.4.2 - 2.4.2) <D16CFDE6-B5F7-301A-995E-8B583D8C675A> /System/Library/Frameworks/OpenCL.framework/Versions/A/OpenCL
    0x7fff97d01000 -     0x7fff97d03ffb  libCGXType.A.dylib (775.16) <B2DC78CA-179F-39A7-8D0B-873DC0ACFE96> /System/Library/Frameworks/CoreGraphics.framework/Versions/A/Resources/libCGXType.A.dylib
    0x7fff97e0c000 -     0x7fff97e20ff7  com.apple.MultitouchSupport.framework (262.33.1 - 262.33.1) <62DF9340-01A1-3E12-A604-C90F6361FD9E> /System/Library/PrivateFrameworks/MultitouchSupport.framework/Versions/A/MultitouchSupport
    0x7fff97e3a000 -     0x7fff97e3bffb  libremovefile.dylib (35) <3485B5F4-6CE8-3C62-8DFD-8736ED6E8531> /usr/lib/system/libremovefile.dylib
    0x7fff97ea1000 -     0x7fff97ea4fff  com.apple.xpc.ServiceManagement (1.0 - 1) <5EFD45BF-B0CD-39F2-8232-6BA33E63E5D4> /System/Library/Frameworks/ServiceManagement.framework/Versions/A/ServiceManagement
    0x7fff97ea8000 -     0x7fff97eafff7  libcompiler_rt.dylib (35) <BF8FC133-EE10-3DA6-9B90-92039E28678F> /usr/lib/system/libcompiler_rt.dylib
    0x7fff97eb0000 -     0x7fff97ecaff7  libextension.dylib (55.1) <6D0CF094-85E8-3F5B-A3F1-25ECF60F80D9> /usr/lib/libextension.dylib
    0x7fff97efe000 -     0x7fff97f39fff  com.apple.Symbolication (1.4 - 56045) <D64571B1-4483-3FE2-BD67-A91360F79727> /System/Library/PrivateFrameworks/Symbolication.framework/Versions/A/Symbolication
    0x7fff98195000 -     0x7fff981b5fff  com.apple.IconServices (47.1 - 47.1) <E83DFE3B-6541-3736-96BB-26DC5D0100F1> /System/Library/PrivateFrameworks/IconServices.framework/Versions/A/IconServices
    0x7fff98693000 -     0x7fff98694fff  liblangid.dylib (117) <B54A4AA0-2E53-3671-90F5-AFF711C0EB9E> /usr/lib/liblangid.dylib
    0x7fff986e7000 -     0x7fff98717fff  com.apple.GSS (4.0 - 2.0) <FD154E62-F4CF-339D-B66C-AF4AED6A94A6> /System/Library/Frameworks/GSS.framework/Versions/A/GSS
    0x7fff98a50000 -     0x7fff98a56ff7  libsystem_networkextension.dylib (167.1.10) <29AB225B-D7FB-30ED-9600-65D44B9A9442> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff98aa5000 -     0x7fff98b19ff3  com.apple.securityfoundation (6.0 - 55126) <DEC91795-7754-334A-8CDA-B429F41B922D> /System/Library/Frameworks/SecurityFoundation.framework/Versions/A/SecurityFoundation

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 1
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 3176955
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=178.1M resident=135.1M(76%) swapped_out_or_unallocated=43.0M(24%)
Writable regions: Total=60.5M written=8064K(13%) resident=10.2M(17%) swapped_out=4K(0%) unallocated=50.3M(83%)
 
REGION TYPE                      VIRTUAL
===========                      =======
CG backing stores                     4K
CG shared images                    176K
Dispatch continuations             8192K
Kernel Alloc Once                     8K
MALLOC                             40.7M
MALLOC (admin)                       32K
Memory Tag 251                        8K
STACK GUARD                        56.0M
Stack                              10.6M
VM_ALLOCATE                        1168K
__DATA                             15.0M
__IMAGE                             528K
__LINKEDIT                         70.3M
__TEXT                            107.8M
__UNICODE                           544K
mapped file                        51.8M
shared memory                         4K
===========                      =======
TOTAL                             362.7M

**Comment by halo** [comment]
Wow! This is certainly helpful thank you very much. I will look into it today. Meanwhile you could try to press "Cancel" really quick in the authorization dialog :) You could also try to turn off Wi-Fi and unplug all Ethernet cables before you start LinkLiar. Not that I have high hopes but then you have at least something to work on while I'm coding along haha :)

**Comment by GnrGnr** [comment]
No difference while network adapters are deactivated when running the installation.

**Comment by kwolk** [comment]
Glad that you have the time now to devote to this project. I too am suffering from the same problem.

**Comment by halo** [comment]
@kwolk So you think I have time for this? ;)

OK, let's give it a new try with the [latest release](https://github.com/halo/LinkLiar/releases/latest), i.e. 1.0.2. Note also the new [debug mode](https://github.com/halo/LinkLiar#troubleshooting)

**Comment by halo** [comment]
@GnrGnr By the way, check out the [backticks ` in Markdown](https://help.github.com/articles/markdown-basics/#multiple-lines) to format logs and that kind of thing. I retrospectively changed your posts to use that formatting.

**Comment by kwolk** [comment]
`Ah, it installs now and appears to be operational!
`It keeps disappearing after selecting an option though, not sure if that's expected behaviour?

**Comment by halo** [comment]
Ah, that's good news :)

Yes, it's expected behavior that the menu disappears after a click and is similar to how every other menu  in the Mac OS top bar works. This is built into it I guess.

**Comment by GnrGnr** [comment]
Installation now works. Yet en1 (WiFi) disappears after authorization of the helper tool from the dropdown. Maybe that's what @halo meant experienced. It doesn't seem to make a difference whether the adapter is active/deativated. Turning wifi on/off also doesn't bring the entry back. En0 works fine though. Like a charm :)

There was some output on the console:

```
16/02/15 22:58:09,467 SecurityAgent[12901]: Failed to initialize IconCache named: com.apple.iconservices with error: Error Domain=NSCocoaErrorDomain Code=4097 "Couldn’t communicate with a helper application." (connection to service named com.apple.iconservices) UserInfo=0x7fa869794d50 {NSDebugDescription=connection to service named com.apple.iconservices}
16/02/15 22:58:09,468 SecurityAgent[12901]: Error returned from iconservicesagent: Error Domain=NSCocoaErrorDomain Code=4097 "Couldn’t communicate with a helper application." (connection to service named com.apple.iconservices) UserInfo=0x7fa8695127e0 {NSDebugDescription=connection to service named com.apple.iconservices}
16/02/15 22:58:13,934 SecurityAgent[12901]: Error returned from iconservicesagent: Error Domain=NSCocoaErrorDomain Code=4097 "Couldn’t communicate with a helper application." (connection to service named com.apple.iconservices) UserInfo=0x7fa869481840 {NSDebugDescription=connection to service named com.apple.iconservices}
16/02/15 22:58:16,016 Link[12873]: <NSXPCConnection: 0x60000010e8e0> connection to service named com.funkensturm.LinkHelper: Warning: Exception caught during decoding of received reply to message 'getVersionWithReply:', dropping incoming message and calling failure block.

Exception: *** -[NSArray objectsAtIndexes:]: index 2 in index set beyond bounds [0 .. 1]
16/02/15 22:58:17,304 Link[12873]: <NSXPCConnection: 0x60000010e8e0> connection to service named com.funkensturm.LinkHelper: Warning: Exception caught during decoding of received reply to message 'getVersionWithReply:', dropping incoming message and calling failure block.

Exception: *** -[NSArray objectsAtIndexes:]: index 2 in index set beyond bounds [0 .. 1]
16/02/15 22:58:19,558 CoreServicesUIAgent[11649]: unexpected message <OS_xpc_error: <error: 0x7fff7abf9c60> { count = 1, contents =
	"XPCErrorDescription" => <string: 0x7fff7abf9f70> { length = 18, contents = "Connection invalid" }
}>
```

Btw: Thanks for the hint to the adequate formatting. I had a quick look yesterday - yet did not identify backticks as the appropriate markdown on first glance.

On another note: Is there any way to shut down LinkLiar except from killing the process/restarting the machine?

**Comment by halo** [comment]
I'll create a separate issue for this because it's not directly related to the installation of the Helper Tool which appears to work for @kwolk now. I can also see in your log that this is something deeper internally which, hopefully, only happens to you :)

Moved over here: https://github.com/halo/LinkLiar/issues/14

Thank you all for your patience and help!
